### PR TITLE
refactor: Share sql lab query

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY
Changed the share SQL Lab Query button into a TypeScript file using react hooks. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
When Query is not saved:
Screen Shot 2021-02-25 at 5 45 38 PM
When Query is saved: 
Screen Shot 2021-02-25 at 5 45 49 PM

### TEST PLAN
re-wrote the tests in react testing library, need some help with the last one. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx [hunk 1]
```diff
@@ -30,17 +29,17 @@ import { storeQuery } from 'src/utils/common';
...
-    dbId: PropTypes.number,
-    title: PropTypes.string,
-    schema: PropTypes.string,
-    autorun: PropTypes.bool,
-    sql: PropTypes.string,
-    remoteId: PropTypes.number,
-  }).isRequired,
-  addDangerToast: PropTypes.func.isRequired,
-};
+interface propTypes {
```
Comment (eschutho) [inline superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx:32]: `ShareSqlLabQueryPropTypes`

File: superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx [hunk 2]
```diff
@@ -30,17 +29,17 @@ import { storeQuery } from 'src/utils/common';
...
+  queryEditor: {
+    dbId: number;
+    title: string;
+    schema: string;
+    autorun: boolean;
+    sql: string;
+    remoteId: number;
```
Comment (eschutho) [inline superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx:39]: looks like this is the only one that is required.

File: superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx [hunk 3]
```diff
@@ -30,17 +29,17 @@ import { storeQuery } from 'src/utils/common';
...
+  };
+  addDangerToast: (msg: string) => void;
```
Comment (eschutho) [inline superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx:41]: oh, and this one. the rest we can mark optional or null

File: superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx [hunk 4]
```diff
@@ -74,28 +66,34 @@ class ShareSqlLabQuery extends React.Component {
...
-      }?savedQueryId=${this.props.queryEditor.remoteId}`;
+      }?savedQueryId=${queryEditor.remoteId}`;
       callback(savedQueryToastContent);
     } else {
       savedQueryToastContent = t('Please save the query to enable sharing');
       callback(savedQueryToastContent);
     }
-  }
+  };
+  const getCopyUrl = (callback: any) => {
```
Comment (eschutho) [inline superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx:87]: the three callbacks can have a function type instead of any

**Review by eschutho — COMMENTED** [review]
looking great! Thanks for tackling these @AAfghahi!

**Comment by eschutho** [comment]
@AAfghahi is this one ready for another look?

**Comment by AAfghahi** [comment]
> @AAfghahi is this one ready for another look?

Yes, it should be.

**Pull request closed without merging**